src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -446,9 +447,15 @@ public void StopStake()
             }
 
             this.stakeCancellationTokenSource.Cancel();
-            this.stakingLoop.Dispose();
+            this.logger.LogTrace("Waiting for staking loop to cancel...");
+            this.stakingLoop?.RunningTask?.GetAwaiter().OnCompleted(() =>
+            {
+                this.logger.LogTrace("Staking loop cancelled.");                              
+            });
+            this.logger.LogTrace("Disposing of staking loop.");
+            this.stakingLoop?.Dispose();
```

Comment:
These 2 are sufficient (together with the cancellation token source questionmark)